global/scripts/mppmodecheck.shlib

```diff
@@ -0,0 +1,231 @@
+# Functions for checking compliance with selected MPP mode for HCP Pipelines
+
+# --- Import log.shlib 
+
+if [[ "$(type -t log_Err_Abort)" != "function" ]]
+then
+    source "$HCPPIPEDIR/global/scripts/log.shlib" "$@"
+fi
+
+
+# --- Functions to check that a valid pipeline name was passed
+
+check_pipeline_name()
+{
+    for valid_pipeline in PreFreeSurfer FreeSurfer FreeSurfer53HCP PostFreeSurfer fMRIVolume fMRISurface Difussion
+    do
+        if [ "${valid_pipeline}" = "${pipeline}" ]; then
+            ispipeline="true"
+        fi
+    done
+}
+
+# --- Functions for pipeline specific checks
+
+check_mpp_PreFreeSurfer()
+{
+    # -- T2w image
+
+    if [ "${T2wInputImages}" = "NONE" ]; then
+      log_Msg "-> T2w image not present (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> T2w image present (HCPStyleData)"
+    fi
+
+    # -- Use of custom brain
+
+    if [ ! "${CustomBrain}" = "NONE" ]; then
+      log_Msg "-> Custom brain used: ${CustomBrain} (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> No custom brain used (HCPStyleData)"
+    fi
+}
+
+check_mpp_FreeSurfer53HCP()
+{
+    # -- T2w image
+
+    if [ "${T2wImage}" = "NONE" ]; then
+      log_Msg "-> T2w image not present (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> T2w image present (HCPStyleData)"
+    fi
+}
+
+check_mpp_FreeSurfer()
+{
+    # -- T2w image
+
+    if [ -z "${p_t2w_image}" ]; then
+        if [ -z "${p_existing_subject}" ]; then
+            log_Msg "-> T2w Image (--t2w-image= or --t2=) not present (LegacyStyleData)"
+            Compliance="LegacyStyleData"
+        fi
+        p_t2w_image="NONE"
+    else
+        log_Msg "T2w Image present: ${p_t2w_image} (HCPStyleData)"
+    fi
+}
+
+
+check_mpp_PostFreeSurfer()
+{
+    # -- T2w image
+
+    if [ ! -e ${T2wFolder}/T2w.nii.gz ]; then
+        log_Msg "-> T2w image not present (LegacyStyleData)"
+        Compliance="LegacyStyleData"
+        T2wRestoreImage="NONE"
+    else
+      log_Msg "-> T2w image present (HCPStyleData)"
+    fi
+}
+
+
+check_mpp_fMRIVolume()
+{
+    # -- Slice timing correction
+
+    if [ ! "${DoSliceTimeCorrection}" == 'NONE' ]; then
+      log_Msg "-> Set to run slice timing correction (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> Not running slice timing correction (HCPStyleData)"
+    fi
+
+    # -- Use of BOLD scout image
+
+    if [ "${fMRIScout}" == 'NONE' ]; then
```

Comment:
I'll have to see the changes in `fMRIVolume`, but `--mpp-mode=HCPStyleData` should already the fMRIScout=NONE condition.  So, this would presumably be changing existing pipeline behavior.

Reply:
I'll remove the check then.